devito/finite_differences/differentiable.py

```diff
@@ -113,6 +113,73 @@ def is_Staggered(self):
     def is_TimeDependent(self):
         return any(i.is_Time for i in self.dimensions)
 
+    def as_independent(self, *deps, **hint):
```

Comment:
This seems quite brutal that we need that almost full copy paste here. Do we know which lines actually cause issue?

Reply:
I think this is resolved, open again if you aren't convinced